### ISSUE TITLE
re-defining a struct can cause it to loose track of its methods

### ISSUE THREAD

**Issue by glycerine** [gijit/gi#20]
At v0.8.8
~~~

// re-defining a struct can cause it to loose
// track of its methods, even just repeating
// a sequence that worked once.

gi> type S struct{}
type S struct{}

elapsed: '78.758µs'
gi> var s S
var s S

elapsed: '33.628µs'
gi>  func (s *S) Hi() {println("S.Hi() called")}
 func (s *S) Hi() {println("S.Hi() called")}

elapsed: '40.708µs'
gi> s.Hi()
s.Hi()
S.Hi() called

elapsed: '70.474µs'
gi> type S struct{}
type S struct{}

elapsed: '213.471µs'
gi> var s S
var s S

elapsed: '47.958µs'
gi>  func (s *S) Hi() {println("S.Hi() called")}
 func (s *S) Hi() {println("S.Hi() called")}

elapsed: '42.912µs'
gi> s.Hi()
s.Hi()
oops: 'problem detected during Go static type checking: 'where error? err = '1:1: invalid operation: s (variable of type S) has no field or method Hi''' on input 's.Hi()'

elapsed: '18.834µs'
gi>  func (s *S) Hi() {println("S.Hi() called")}
 func (s *S) Hi() {println("S.Hi() called")}

elapsed: '40.92µs'
gi> s.Hi()
s.Hi()
oops: 'problem detected during Go static type checking: 'where error? err = '1:1: invalid operation: s (variable of type S) has no field or method Hi''' on input 's.Hi()'

elapsed: '37.28µs'
gi> var s S
var s S

elapsed: '23.326µs'
gi> s.Hi()
s.Hi()
oops: 'problem detected during Go static type checking: 'where error? err = '1:1: invalid operation: s (variable of type S) has no field or method Hi''' on input 's.Hi()'

elapsed: '18.826µs'
gi> 
~~~

**Comment by glycerine** [comment]
red test 001 in cmd/gi/replay_test.go

**Comment by glycerine** [comment]
fixed in v0.9.6, by having types/check.go call check.deleteFromObjMapPriorTypeName(oname) when oname is the obj.Name() for a TypeName that is being re-defined.
